### DISCUSSION TITLE
Reloading when clicking a link to the same page

### DISCUSSION THREAD

**Discussion by StephenCoventry** [swup/swup#483] [Help] [answered]
Is possible to reload the same page when a link with the same URL is clicked? It looks like the standard behaviour is to do nothing if the URL matches but this is a little confusing for users if it isn't obvious that the link is to the same page.

**Comment by StephenCoventry** [comment, marked as answer]
**Update**: [swup 4](https://swup.js.org/announcements/swup-4/) supports defining custom logic for internal hooks. You can make swup follow links to the current page by replacing the internal handler of the `link:self` hook:

```js
swup.hooks.replace('link:self', (visit) => swup.navigate(visit.to.url))
```

In swup 3, this works as well:

```js
swup.on("samePage", () => swup.loadPage({ url: window.location.href }))
```

Ok, so I just found the `samePage` event and using that I can get fire the `loadPage` method which does what I want, but is there a 'better' way?

**Reply by daun** [reply]
Yes, that's exactly how to do it. You could reset window.location, but then you're skipping the animation and triggering a hard reload.
